jest.config.js

```diff
@@ -0,0 +1,29 @@
+// jest.config.js
+const nextJest = require('next/jest');
+
+const createJestConfig = nextJest({
+  // Provide the path to your Next.js app to load next.config.js and .env files in your test environment
+  dir: './'
+});
+
+// Add any custom config to be passed to Jest
+/** @type {import('jest').Config} */
+const customJestConfig = {
+  // Add more setup options before each test is run
+  // setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
+  // if using TypeScript with a baseUrl set to the root directory then you need the below for alias' to work
+  moduleDirectories: ['node_modules', '<rootDir>/'],
+
+  // If you're using [Module Path Aliases](https://nextjs.org/docs/advanced-features/module-path-aliases),
+  // you will have to add the moduleNameMapper in order for jest to resolve your absolute paths.
+  // The paths have to be matching with the paths option within the compilerOptions in the tsconfig.json
+  // For example:
+
+  moduleNameMapper: {
+    '@/(.*)$': '<rootDir>/src/$1'
+  },
```

Comment:
```suggestion
```
I would recommend not adding custom configuration until you need it.  It has a way of sticking around and confusing future maintainers (who will assume it's necessary).

Reply:
Have to have this config for integration with nextjs. If we just straight remove it we get errors like
`Support for the experimental syntax 'jsx' isn't currently enabled`

I removed a bunch of the unnecessary things we had in it however.

Reply:
The config does need exist, but `moduleDirectories` does not need to be customised (the default, `node_modules`, is all you need).

`testEnvironment: 'jest-environment-jsdom'` is fine, though.

Reply:
So, to confirm, the `customJestConfig` would only need to include `testEnvironment: 'js-environment-jsdom`, correct?

Reply:
Yep, that's what I found.